verilog/tools/ls/symbol-table-handler.h

```diff
@@ -0,0 +1,57 @@
+#ifndef VERILOG_TOOLS_LS_SYMBOL_TABLE_HANDLER_H
+#define VERILOG_TOOLS_LS_SYMBOL_TABLE_HANDLER_H
+
+#include <memory>
+#include <unordered_set>
+#include <vector>
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "common/lsp/lsp-protocol.h"
+#include "verilog/analysis/symbol_table.h"
+#include "verilog/analysis/verilog_project.h"
+
+namespace verilog {
+
+// A class interfacing the SymbolTable with the LSP messages.
+// It manages the SymbolTable and its necessary components,
+// and provides such information as symbol definitions
+// based on LSP requests.
+// The provided information is in LSP-friendly format.
+class SymbolTableHandler {
+ public:
+  SymbolTableHandler(){};
+
+  // Sets the project for the symbol table.
+  // VerilogProject requires root, include_paths and corpus to
+  // create a base of files that may contain definitions for symbols.
+  // Once the project's root is set, a new SymbolTable is created.
+  void setProject(absl::string_view root,
+                  const std::vector<std::string> &include_paths,
+                  absl::string_view corpus);
+
+  // Creates a new symbol table given the VerilogProject in setProject
+  // method.
+  void resetSymbolTable();
+
+  // Fills the symbol table for a given verilog source file.
+  void buildSymbolTableFor(VerilogSourceFile &file);
+
+  // Finds the definition for a symbol provided in the DefinitionParams
+  // message delivered i.e. in textDocument/definition message.
+  // Provides a list of locations with symbol's definitions.
+  std::vector<verible::lsp::Location> findDefinition(
+      const verible::lsp::DefinitionParams &params);
+
+ private:
+  std::shared_ptr<VerilogProject> currproject =
+      nullptr;  // current VerilogProject for which the symbol table is created
+  std::shared_ptr<SymbolTable> symboltable = nullptr;  // symbol table structure
+  std::unordered_set<std::string>
```

Comment:
prefer absl:: type containers as they are generally faster and less memory intensive.
`absl::flat_hash_set` would be a good choice here.

Reply:
OK, I'll check in future development for `absl` equivalents for containers